src/content/app/genome-browser/state/browser-location/browserLocationSlice.ts

```diff
@@ -0,0 +1,173 @@
+/**
+ * See the NOTICE file distributed with this work for additional information
+ * regarding copyright ownership.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import {
+  Action,
+  ActionCreator,
+  createSlice,
+  Dispatch,
+  PayloadAction,
+  ThunkAction
+} from '@reduxjs/toolkit';
+import { replace } from 'connected-react-router';
+import { isEqual } from 'lodash';
+import pickBy from 'lodash/pickBy';
+
+import browserStorageService from 'src/content/app/genome-browser/services/browser-storage-service';
+import * as urlFor from 'src/shared/helpers/urlHelper';
+
+import { getChrLocationStr } from 'src/content/app/genome-browser/helpers/browserHelper';
+import { ParsedUrlPayload } from 'src/content/app/genome-browser/state/browser-entity/browserEntitySlice';
+import { buildFocusIdForUrl } from 'src/shared/state/ens-object/ensObjectHelpers';
+
+import { getChrLocation } from 'src/content/app/genome-browser/state/browser-location/browserLocationSelectors';
+import {
+  getBrowserActiveGenomeId,
+  getBrowserActiveEnsObjectId
+} from 'src/content/app/genome-browser/state/browser-entity/browserEntitySelectors';
+
+import { RootState } from 'src/store';
+
+const isServer = typeof window === 'undefined';
+const chrLocations = !isServer ? browserStorageService.getChrLocation() : [];
+
+export type ChrLocation = [string, number, number];
+
+export type ChrLocations = { [genomeId: string]: ChrLocation };
+
+export type BrowserLocationState = Readonly<{
+  chrLocations: ChrLocations; // final location of the browser when user stopped dragging/zooming; used to update the url
+  actualChrLocations: ChrLocations; // transient locations that change while user is dragging or zooming
+  regionEditorActive: boolean;
+  regionFieldActive: boolean;
+  isObjectInDefaultPosition: boolean;
+}>;
+
+export const defaultBrowserLocationState: BrowserLocationState = {
+  chrLocations,
+  actualChrLocations: {},
+  regionEditorActive: false,
+  regionFieldActive: false,
+  isObjectInDefaultPosition: false
+};
+
+export const setActualChrLocation: ActionCreator<
+  ThunkAction<any, any, null, Action<string>>
+> = (chrLocation: ChrLocation) => {
+  return (dispatch: Dispatch, getState: () => RootState) => {
```

Comment:
Here and below; we shouldn't need to specify the `Dispatch` type

Reply:
That's done